crates/sui-json-rpc-types/src/sui_transaction.rs

```diff
@@ -1814,6 +1848,37 @@ impl SuiPureValue {
     }
 }
 
+impl Display for SuiPureValue {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        writeln!(f, " ┌── PureValue")?;
+        if let Some(t) = self.value_type() {
+            writeln!(f, " │ Type: {t}")?;
+        };
+        // try to convert a bcs array into a Base64 string
+        let mut json_value = self.value.to_json_value();
+        bytes_array_to_base64(&mut json_value);
+
+        // if we have a Base64 string, that's likely going to be very long
+        // thus we need to split it by the terminal's width
+        // minus some small number of characters (left + right padding)
+        let width = get_terminal_width();
+        let mut value = json_value.to_string();
+        if value.len() > width - 15 {
```

Comment:
That makes sense. However, where would one call the `get_terminal_width`? Wouldn't it in the end be called within a display?

Reply:
We discussed this offline, but recapping here:

The idea would be to call it outside of `Display` to create one of these wrapping types with the appropriate `width` field filled out (i.e. this width field is not the terminal width, but the width you are allowing for this field to be wrapped in, so it will have any padding etc accounted for).

You may need to introduce multiple of these wrapping types as well, so that each one accounts for its own padding etc before delegating to a wrapped display for its inner fields.